src/Sylius/Component/Product/Generator/SlugGenerator.php

```diff
@@ -22,9 +22,8 @@ final class SlugGenerator implements SlugGeneratorInterface
      */
     public function generate(string $name): string
     {
-        // Manually replacing apostrophes since Transliterator started removing them at v1.2.
-        $name = str_replace('\'', '-', $name);
+        $unicodeString = (new AsciiSlugger())->slug($name);
```

Comment:
Could be an one-liner:

```suggestion
        return (new AsciiSlugger())->slug($name)->lower()->toString();
```